architecture/2024-08-applicationdeployment-controller.md

```diff
@@ -0,0 +1,370 @@
+
+<!--
+Provide a succinct high-level description of the component or feature and 
+where/how it fits in the big picture. The overview should be one to three 
+paragraphs long and should be understandable by someone outside the Radius
+team. Do not provide the design details in this, section - there is a
+dedicated section for that later in the document.
+-->
+
+Currently, there is little support for deploying Radius resources without the use of the Radius CLI. This complicates future scenarios (such as support for GitOps) because GitOps and other tools that we want to integrate with are unable to deploy Radius resources (as defined in Bicep manifests) directly. Today, Radius is primarily deployed onto Kubernetes clusters, and naturally most of the existing Kubernetes ecosystem only works with Kubernetes resources. This design document proposes updates to Radius will allow users to deploy Radius resources by authoring Kubernetes CRDs.
+
+## Terms and definitions
+
+<!--
+Include any terms, definitions, or acronyms that are used in
+this design document to assist the reader. They may or may not
+be part of the user-facing experience once implemented, and can
+be specific to this design context.
+-->
+
+**CRD (Custom Resource Definition)**: A Kubernetes resource that allows users to define their own resource types.
+
+**GitOps** A way to do Kubernetes cluster management and application delivery. It works by using Git as a single source of truth for declarative infrastructure and applications.
+
+## Objectives
+
+<!--
+Describe goals/non-goals and user-scenario of this feature to understand
+the end-user goals.
+* If the feature shares the same objectives of the existing design, link
+  to the existing doc and section rather than repeat the same context.
+* If the feature has a scenario, UX, or other product feature design doc,
+  link it here and summarize the important parts.
+-->
+
+### Goals
+
+<!--
+Describe goals to define why we are doing this work, how we will make
+priority decisions, and how we will determine success.
+-->
+
+**Goal: Radius resources can be deployed by authoring a Kubernetes CRD**
+
+**Goal: Radius resources can be deployed without using the Radius CLI or directly calling Radius APIs**
+
+**Users have a simple script available to generate the necessary CRD**
+
+### Non-goals
+
+<!--
+Describe non-goals to identify something that we won’t be focusing on 
+immediately. We won’t be expending any effort on these matters. If there
+will be follow-ups after this work, list them here. If there are things
+we plan to do in the future, but are out of scope of this design, list
+them here. Provide a brief explanation on why this is a non-goal.
+-->
+
+**Non-goal: Full support for GitOps**
+- We will not yet be implementing automatic generation of ApplicationDeployment resources from Bicep manifests or querying Git repositories. This design will be covered in a separate design document.
+
+**Non-goal: Deletion of resources**
```

Comment:
I'd suggest we bring this in. I think it's going to be more complicated to leave out.